ChurchCRM/CRM
# Improve setup database checks

#### What's this PR do?
*  Adds a real-time database connection validation to the setup wizard.
*  improves error handling in the bootstrapper
*  Adds database setup errors to the log output (app.log)

#### Screenshots (if appropriate)
![Setup Wizard DB Check](https://user-images.githubusercontent.com/11679900/83368105-a4dac500-a385-11ea-8ace-004a27c5ce05.gif)

#### What Issues does it Close?

Relates to #5243 #5244 

#### Any background context you want to provide?
Sometimes new installs would silently fail if the database user does not have `CREATE VIEW` SQL Permission.  This makes those failures non-silent.


#### How should this be manually tested?

Remove your `Include/Config.php` file to trigger the setup wizard.
Provide a MySQL user with appropriate permissions - ensure correct setup and schema install.

Repeat above, but with a user having less-than-complete permissions (as defiend in https://github.com/ChurchCRM/CRM/pull/5253/files#diff-5ce6b7a7db09f43ffe166bbb62ca2849R40): `"SELECT,INSERT,UPDATE,DELETE,CREATE,DROP,ALTER,CREATE VIEW"`
Ensure that the setup page warns you about incomplete permissions.
